src/theme/DocItem/Metadata/determineCanonical.js

```diff
@@ -107,7 +107,7 @@ function determineCanonicalFromDoc(currentDoc, currentPlugin) {
     .find((doc) => doc.id === unversionedId);
 
   if (match) {
-    if (match.path.match(/(?<=(optimize|docs)\/)((next|[0-9\.]*)\/)(?=.+)/)) {
+    if (/(optimize|docs)((next|[0-9\.]*)\/)/.test(match.path)) {
```

Comment:
This test is _also_ only used to emit a warning message to the console, which is more useful during the server build than when viewing the page in a browser. 

I'd imagine docusaurus exposes a context that tells us if we are operating on the server or browser. If you'd prefer, I can investigate this and add more guarding against emitting this message in the browser.

Reply:
I don't have strong feelings about the additional investigation.